client/pages/events/index.tsx

```diff
@@ -1,9 +1,35 @@
 import React from 'react'
 import { EventViewToggle } from '../../components/events'
 import { useAppContext } from '../../context'
+import { Event } from '../../context'
+
+export const EVENT_MAP_KEY_FORMAT = 'YYYY-MM-DD'
+
+export interface FilteredEvents {
+  filteredEventsDateMap: Map<string, Array<Event>>
+  dateOrder: Array<string>
+}
 
 export default function Events() {
   const { events } = useAppContext()
-  const filteredEvents = Array.from(events.values())
+  const filteredEventsArray = Array.from(events.values())
+
+  //this requires that filteredEventsArray is in sorted order (with respect to event startDate)
+  const filteredEventsDateMap: Map<string, Array<Event>> = new Map()
+  const dateOrder: Array<string> = []
```

Comment:
This is just a list of keys for the map?

I think we can avoid storing `dateOrder` altogether and just let the `EventViewToggle` component call `Array.from(filteredEventsDateMap.keys()).sort()` to get a guaranteed sorted list of keys for the map.

That way, we can also relax the requirement that `filteredEventsArray` is sorted. It would also do away with the `FilteredEvents` interface, since we'd be dealing with a single object (the map) again.